### PR TITLE
Add additional validation to the webhook.

### PR DESCRIPTION
**Pull request by respringer**

### CODE REVIEW

File: operator/pkg/apis/cassandra/v1beta1/webhook.go [hunk 1]
```diff
@@ -5,35 +5,109 @@ package v1beta1
...
+		return fmt.Errorf("CassandraDatacenter attempted to change StorageConfig")
+	}
+
+	// Topology changes - Racks
+	// - Rack Name and Zone changes are disallowed.
+	// - Removing racks is not supported.
+	// - Reordering the rack list is not supported.
+	// - Any new racks must be added to the end of the current rack list.
+
+	if len(oldDc.Spec.Racks) > len(newDc.Spec.Racks) {
```
Comment (jimdickinson) [inline operator/pkg/apis/cassandra/v1beta1/webhook.go:64]: sadly, we have to add the calls to GetRacks for both of these, I think

File: operator/pkg/apis/cassandra/v1beta1/webhook.go [hunk 2]
```diff
@@ -5,35 +5,109 @@ package v1beta1
...
+	if dc.Spec.ServerType == "cassandra" && dc.Spec.ServerVersion != "3.11.6" && dc.Spec.ServerVersion != "4.0.0" {
+		return fmt.Errorf("CassandraDatacenter attempted to use unsupported Cassandra version '%s'",
+			dc.Spec.ServerVersion)
+	}
+
+	return nil
+}
+
+// Ensure that no values are improperly set
+func ValidateDatacenterFieldChanges(oldDc CassandraDatacenter, newDc CassandraDatacenter) error {
```
Comment (jimdickinson) [inline operator/pkg/apis/cassandra/v1beta1/webhook.go:35]: great signature here and above

File: operator/pkg/apis/cassandra/v1beta1/webhook.go (hunk 1 above)
Comment (respringer) [reply operator/pkg/apis/cassandra/v1beta1/webhook.go:64]: Is this all to support the special case of the user not specifying any racks?

Comment (jimdickinson) [reply operator/pkg/apis/cassandra/v1beta1/webhook.go]: that's right

**Pull request merged**